test/e2e/e2ethanos/services.go

```diff
@@ -345,6 +345,9 @@ func (q *QuerierBuilder) collectArgs() ([]string, error) {
 
 		args = append(args, "--store.sd-files="+filepath.Join(q.InternalDir(), "filesd.yaml"))
 	}
+
+	args = append(args, "--query.active-query-path="+q.InternalDir())
```

Comment:
So this breaks a test, specifically the `TestQueryCompatibilityWithPreInfoAPI` e2e test, reason being that we use specific images for the test cases on that one [here](https://github.com/thanos-io/thanos/blob/main/test/e2e/query_test.go#L383). The new `-query.active-query-path` flag won't exist on any image but the latest, so this cannot be an always-enabled option. 🙂 

Maybe you can address @GiedriusS's comment and make the flag optional, and add a new `WithActiveQueryTracker` method on `QuerierBuilder` and use that instead in a dedicated e2e test which checks for this file?

Reply:
So from my attempts at an E2E test, I _think_ the same issues that plagued this PR during CI will occur during an E2E test. The issue I'm referring to is related to container permissions and seems to be a known issue: https://github.com/prometheus/prometheus/issues/5976.
Is there any alternative way I could try?

Reply:
After making this flag optional in E2E and add WithActiveQueryTracker then you should be good.

Reply:
Yes, what I think you could do is maybe execute a command, to change permissions for the data dir, as mentioned in [this comment](https://github.com/prometheus/prometheus/issues/5976#issuecomment-772299233)! 🙂

Reply:
It might be a bit too difficult to make it work in an e2e setup, so I think it would be fine to skip an e2e test for this. But maybe you can document this issue in Querier docs instead? So users are aware of the implications of using this! 🙂